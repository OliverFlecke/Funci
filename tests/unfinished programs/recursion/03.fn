--
-- until p f  yields the result of applying f until p holds.
--
-- (slow in the interpreter)
--

main :: Int = until (longtime) (inc) 0;

inc x :: (Int -> Int) = x + 1;
longtime i :: (Int -> Bool) = 100 < i;

until p f x :: ((Int -> Bool) -> (Int -> Int) -> Int -> Int) =
                if (p x) 
                    then x 
                    else until p f (f x) 