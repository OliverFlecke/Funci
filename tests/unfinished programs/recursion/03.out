Number (I 100001) (Unit [])